contracts/modules/Offers/V1/OffersV1.sol

```diff
@@ -0,0 +1,422 @@
+    //     |   createOffer()  |                              |
+    //     | ----------------->                              |
+    //     |                  |                              |
+    //     |                  |        transferFrom()        |
+    //     |                  | ----------------------------->
+    //     |                  |                              |
+    //     |                  |                              |----.
+    //     |                  |                              |    | transfer tokens into escrow
+    //     |                  |                              |<---'
+    //     |                  |                              |
+    //     |                  |----.                         |
+    //     |                  |    | offer count ++          |
+    //     |                  |<---'                         |
+    //     |                  |                              |
+    //     |                  |----.                         |
+    //     |                  |    | create offer            |
+    //     |                  |<---'                         |
+    //     |                  |                              |
+    //     |                  |----.
+    //     |                  |    | offersFor[NFT].append(id)
+    //     |                  |<---'
+    //     |                  |                              |
+    //     |                  |----.                         |
+    //     |                  |    | emit OfferCreated()     |
+    //     |                  |<---'                         |
+    //     |                  |                              |
+    //     |        id        |                              |
+    //     | <-----------------                              |
+    //   Caller           ,---+----.               ,---------+---------.
+    //     ,-.            |OffersV1|               |ERC20TransferHelper|
+    //     `-'            `--------'               `-------------------'
+    //     /|\
+    //      |
+    //     / \
+    /// @notice Creates an offer for an NFT
+    /// @param _tokenContract The address of the desired ERC-721 token
+    /// @param _tokenId The ID of the desired ERC-721 token
+    /// @param _currency The address of the ERC-20 token offering, or address(0) for ETH
+    /// @param _amount The amount offering
+    /// @param _findersFeeBps The bps of the amount (post-royalties) to send to a referrer of the sale
+    /// @return The ID of the created offer
+    function createOffer(
+        address _tokenContract,
+        uint256 _tokenId,
+        address _currency,
+        uint256 _amount,
+        uint16 _findersFeeBps
+    ) external payable nonReentrant returns (uint256) {
+        require(_findersFeeBps <= 10000, "createOffer finders fee bps must be less than or equal to 10000");
+
+        // Validate offer and take custody
+        _handleIncomingTransfer(_amount, _currency);
+
+        offerCount++;
```

Comment:
can make this unchecked, the sun will devour the earth before it could ever overflow

Reply:
LOL ty king